src/plonkup/multiset.rs

```diff
@@ -100,10 +100,17 @@ impl MultiSet {
     /// look as follows, s: {1,1,2,2,3,3,4,4}
     pub fn sorted_concat(&self, f: &MultiSet) -> Result<MultiSet, Error> {
         let mut s = self.clone();
-        s.0.reserve(f.0.len());
-        for element in f.0.iter() {
-            let index = s.position(element).ok_or(Error::ElementNotIndexed)?;
-            s.0.insert(index, *element);
+        // s.0.reserve(f.0.len());
+        // for element in f.0.iter() {
+        //     let index = s.position(element).ok_or(Error::ElementNotIndexed)?;
+        //     s.0.insert(index, *element);
+        // }
+        s.0.extend_from_slice(&f.0);
+        let mut s_normal = s.0.iter().map(|x| x.reduce()).collect::<Vec<_>>();
+        s_normal.sort();
+        let mut s = MultiSet::new();
+        for element in s_normal.iter() {
+            s.push(BlsScalar::from_raw(element.0));
```

Comment:
maybe instead of extend, collect, sort, push; we might use itertools so we don't build a vec

this means we can sort the iterator directly with [itertools::sorted](https://docs.rs/itertools/0.10.3/itertools/fn.sorted.html) and then collect directly into a `MultiSet`